### DISCUSSION TITLE
module type not available in esp32

### DISCUSSION THREAD

**Discussion by bkbartk** [arendst/Tasmota#15397] [Q&A] [answered]
for my wemos mini esp32 I need to select Module type "18 generic"
https://tasmota.github.io/docs/devices/Wemos-D1-Mini/
but I only have esp32-devkit
the issue is that GPIO 12/13/14
should be marked as D6/D7/D5

But now they are marked as some generic "IO"
I think this is why my config fails..
How can I apply this template to ESP32?

I tried some stuff but I can't find it out,
simply applying this template doesn't work.

**Comment by barbudor** [comment]
18 Generic is only for ESP8266
The equivalent for ESP32 is indeed ESP32-DevKit

What are you trying to achieve ?

**Reply by sfromis** [reply]
ESP32-DevKit is module number 1 on ESP32. As you see, there is no attempt to match the varied ways the pins on ESP32 modules are marked. On most modules, the numbers in the labels match the GPIO numbers for the pins. If/when they do not, you'll have to manage :-)

This is unrelated to a config failing or not. What template are you having trouble applying? The Wemos D1 Mini page you linked is unrelated to ESP32 boards.

**Reply by barbudor** [reply]
You can use the below diagram to match GPIOs and Dx

![image](https://user-images.githubusercontent.com/2996042/163874970-3ffef3e2-55a7-417a-ae5c-7afd9eda5785.png)

**Comment by bkbartk** [comment, marked as answer]
I have this screenshot of settings
![image](https://user-images.githubusercontent.com/17070805/163874578-85325e05-9d84-40e5-b829-c4442d8f82df.png)
And I want to apply those to a ESP32 wemos
even when I try to map the GPIO, which might or might not be different
![image](https://user-images.githubusercontent.com/17070805/163874984-ecc60dda-36dc-4832-a65d-e80aaf034d74.png)
tasmota never shows the temperatur sensor
![image](https://user-images.githubusercontent.com/17070805/163875068-bb7ee10d-1029-400d-b66b-74da662bdd70.png)
There is a faulty reading, which makes sence because I haven't attached everything yet, But I would expect a temparature reading. and if some wire fails, I would expect the sensor to be in the overview with an error or something.

Edit: ok, clear, I should do the mapping,
but with the mapping, I still expect a temperature to be shown.

**Reply by barbudor** [reply]
If there is no DS18x20 connected, Tasmota will report nothing
Same for the OpenTherm, if the actual hardware is not connected, the fault is normal

**Reply by sfromis** [reply]
Forget about labels on the ESP8266 board, ESP32 is a different thing. For reference:
![image](https://user-images.githubusercontent.com/47082390/163875511-c9ad4084-c882-4aae-b9cd-49b86ab57493.png)

**Reply by sfromis** [reply]
Note that DS18B20 requires an external pullup resistor on ESP32, you cannot enable internal pullup.

**Comment by bkbartk** [comment]
I tried another opentherm shield (bought 3) , And this one actually shows me a temperature And the 3th one showed me more opentherm information.

So the issue is not in tasmota, but probably somewhere in the wiring. For testing I just used pinheaders, instead of actual soldering. I expected some faulty reads. but at lease that missing elements were shown in tasmota layout

**Reply by sfromis** [reply]
What do you mean by "missing elements were shown in tasmota layout"? In many cases, Tasmota will silently ignore pins if it can't get input from a device, in a few cases something bad will be shown, but in neither case there will be some sort of diagnostic.

Of course, you should ensure correct wiring, and matching gpio config, with reference to the diagram above.

**Reply by bkbartk** [reply]
I mean the ignore part, I would have expected always to see something like "Error/bad/unavailable" But now I know it doesn't it's not an issue anymore.
GPIO config is ok, but I still need to work on my wiring. This might take a while because this is not the part I'm good at.
